llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1406,6 +1405,16 @@ PreservedAnalyses AddressSanitizerPass::run(Module &M,
       ClUseStackSafety ? &MAM.getResult<StackSafetyGlobalAnalysis>(M) : nullptr;
 
   if (Triple(M.getTargetTriple()).isSPIR()) {
+    for (Function &F : M) {
+      // ESIMD kernel doesn't support noinline functions, so we can't
+      // support sanitizer for it
+      if (F.hasMetadata("sycl_explicit_simd")) {
+        F.removeFnAttr(Attribute::SanitizeAddress);
```

Comment:
Is it necessary to remove the attribute?
If there are multiple esimd functions, does it make sense to just remove the attribute for the first one?

Reply:
Okay, I think I need to remove all "Attribute::SanitizeAddress" in esimd function and then exit.

Reply:
Done